builtin/providers/aws/resource_aws_api_gateway_integration_response.go

```diff
@@ -0,0 +1,153 @@
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem:     schema.TypeString,
+			},
+		},
+	}
+}
+
+func resourceAwsApiGatewayIntegrationResponseCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+
+	templates := make(map[string]string)
+	for k, v := range d.Get("response_templates").(map[string]interface{}) {
+		templates[k] = v.(string)
+	}
+
+	_, err := conn.PutIntegrationResponse(&apigateway.PutIntegrationResponseInput{
+		HttpMethod:         aws.String(d.Get("http_method").(string)),
+		ResourceId:         aws.String(d.Get("resource_id").(string)),
+		RestApiId:          aws.String(d.Get("api_id").(string)),
+		StatusCode:         aws.String(d.Get("status_code").(string)),
+		ResponseTemplates:  aws.StringMap(templates),
+		ResponseParameters: nil,
+	})
+	if err != nil {
+		return fmt.Errorf("Error creating API Gateway Integration Response: %s", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s-%s-%s-%s", d.Get("api_id").(string), d.Get("resource_id").(string), d.Get("http_method").(string), d.Get("status_code").(string)))
+	log.Printf("[DEBUG] API Gateway Integration Response ID: %s", d.Id())
+
+	return nil
+}
+
+func resourceAwsApiGatewayIntegrationResponseRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+
+	log.Printf("[DEBUG] Reading API Gateway Integration Response %s", d.Id())
+	out, err := conn.GetIntegrationResponse(&apigateway.GetIntegrationResponseInput{
+		HttpMethod: aws.String(d.Get("http_method").(string)),
+		ResourceId: aws.String(d.Get("resource_id").(string)),
+		RestApiId:  aws.String(d.Get("api_id").(string)),
+		StatusCode: aws.String(d.Get("status_code").(string)),
+	})
+	if err != nil {
+		return err
+	}
+	log.Printf("[DEBUG] Received API Gateway Integration Response: %s", out)
+	d.SetId(fmt.Sprintf("%s-%s-%s-%s", d.Get("api_id").(string), d.Get("resource_id").(string), d.Get("http_method").(string), d.Get("status_code").(string)))
+
+	return nil
+}
+
+func resourceAwsApiGatewayIntegrationResponseUpdate(d *schema.ResourceData, meta interface{}) error {
+	return resourceAwsApiGatewayIntegrationResponseCreate(d, meta)
+}
+
+func resourceAwsApiGatewayIntegrationResponseDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+	log.Printf("[DEBUG] Deleting API Gateway Integration Response: %s", d.Id())
+
+	resourceId := d.Get("resource_id").(string)
+	if o, n := d.GetChange("resource_id"); o.(string) != n.(string) {
+		resourceId = o.(string)
+	}
+	httpMethod := d.Get("http_method").(string)
+	if o, n := d.GetChange("http_method"); o.(string) != n.(string) {
+		httpMethod = o.(string)
+	}
+	restApiID := d.Get("api_id").(string)
+	if o, n := d.GetChange("api_id"); o.(string) != n.(string) {
+		restApiID = o.(string)
+	}
+	statusCode := d.Get("status_code").(string)
+	if o, n := d.GetChange("status_code"); o.(string) != n.(string) {
+		statusCode = o.(string)
+	}
```

Comment:
This is an interesting piece of code above :thought_balloon: Can you explain what is the motivation behind it?

Is it because you're worried about user modifying parameters & subsequently running `destroy` and terraform not being able to delete the resource as these would differ?